Mathlib/Probability/ConditionalProbability.lean

```diff
@@ -251,4 +253,67 @@ lemma sum_meas_smul_cond_fiber {X : Ω → α} (hX : Measurable X) (μ : Measure
       rw [← measure_biUnion_finset _ fun _ _ ↦ (hX (.singleton _)).inter hE, this]
       aesop (add simp [PairwiseDisjoint, Set.Pairwise, Function.onFun, disjoint_left])
 
+end Fintype
+
+variable {ι Ω α β : Type*} {mΩ : MeasurableSpace Ω} {mα : MeasurableSpace α}
+  {mβ : MeasurableSpace β} {μ : Measure Ω} {X : ι → Ω → α} {Y : ι → Ω → β} {f : _ → Set Ω}
+  {t : ι → Set β} {s : Finset ι}
+
+/-- The probability of an intersection of preimages conditioning on another intersection factors
+into a product. -/
+lemma cond_iInter [Finite ι] (hY : ∀ i, Measurable (Y i))
+    (hindep : iIndepFun (fun _ ↦ mα.prod mβ) (fun i ω ↦ (X i ω, Y i ω)) μ)
+    (hf : ∀ i ∈ s, MeasurableSet[mα.comap (X i)] (f i))
+    (hy : ∀ i, μ (Y i ⁻¹' t i) ≠ 0) (ht : ∀ i, MeasurableSet (t i)) :
+    μ[|⋂ i, Y i ⁻¹' t i] (⋂ i ∈ s, f i) = ∏ i ∈ s, μ[|Y i in t i] (f i) := by
```

Comment:
```suggestion
    μ[⋂ i ∈ s, f i |⋂ i, Y i ⁻¹' t i] = ∏ i ∈ s, μ[f i|Y i in t i] := by
```
Does this work?

Reply:
Whoops sorry I had already done this change locally, but somehow reverted it before pushing